message_center_compassion/model/child_compassion.py

```diff
@@ -72,144 +77,134 @@ def allocate(self, cr, uid, args, context=None):
 
         if child.state == 'E':
             # Put the child back to normal state
-            self.reset_child_state(cr, uid, child.id, context)
+            child.reset_child_state()
 
         return True
 
-    def reset_child_state(self, cr, uid, ids, context=None):
-        if not isinstance(ids, list):
-            ids = [ids]
-        for child in self.browse(cr, uid, ids, context):
+    def reset_child_state(self):
+        for child in self:
             child.write({'state': child.previous_state,
                          'previous_state': False})
         return True
 
-        ######################################################################
-        #                          PRIVATE METHODS                           #
-        ######################################################################
-
-    def _insert_new_child(self, cr, uid, vals, context=None):
-        """Creates a non-existing child and link messages
-        referencing it."""
-        child_id = self.create(cr, uid, vals, context)
-        # Link existing messages to the child
-        mess_obj = self.pool.get('gmc.message.pool')
-        mess_ids = mess_obj.search(
-            cr, uid, [('incoming_key', '=', vals.get('code')),
-                      ('object_id', '=', 0)],
-            context=context)
-        if mess_ids:
-            mess_obj.write(cr, uid, mess_ids, {
-                'object_id': child_id,
-                'child_id': child_id}, context)
-        return child_id
-
-    def view_error_messages(self, cr, uid, ids, context=None):
+    ##########################################################################
+    #                             VIEW CALLBACKS                             #
+    ##########################################################################
+    @api.multi
+    def view_error_messages(self):
         return {
             'type': 'ir.actions.act_window',
             'id': 'action_gmc_message_incoming_form',
             'view_mode': 'tree,form',
             'res_model': 'gmc.message.pool',
-            'domain': [('child_id', 'in', ids), ('state', '=', 'failure')]
+            'domain': [('child_id', 'in', self.ids),
+                       ('state', '=', 'failure')]
         }
+
+    ##########################################################################
+    #                             PRIVATE METHODS                            #
+    ##########################################################################
+    def _insert_new_child(self, vals):
+        """Creates a non-existing child and link messages
+        referencing it."""
+        child_id = self.create(vals).id
+        # Link existing messages to the child
+        messages = self.env['gmc.message.pool'].search([
+            ('incoming_key', '=', vals.get('code')),
+            ('object_id', '=', 0)])
+        messages.write({
+            'object_id': child_id,
+            'child_id': child_id})
+        return child_id

```

Comment:
In this case, the method is only called from this file and uses only the id outside. But it's true the standard is more to return the record. Do you think we should always return Records or is it ok to have some exceptions ?